generators/messaging-adapter-generator/src/main/java/uk/gov/justice/raml/jms/core/JmsEndpointGenerator.java

```diff
@@ -45,7 +51,13 @@ public void run(final Raml raml, final GeneratorConfig configuration) {
 
         raml.getResources().values().stream()
                 .filter(resource -> resource.getAction(POST) != null)
-                .map(resource -> messageListenerCodeGenerator.generateFor(resource, new BaseUri(raml.getBaseUri())))
+                .flatMap(resource -> {
+                    final BaseUri baseUri = new BaseUri(raml.getBaseUri());
+                    final Stream<TypeSpec> messageListenerGenerator = Stream.of(messageListenerCodeGenerator.generatedCodeFor(resource, baseUri));
+                    return baseUri.component() == EVENT_LISTENER && !containsGeneralJsonMimeType(resource.getActions())
+                            ? concat(messageListenerGenerator, Stream.of(eventFilterCodeGenerator.generatedCodeFor(resource, baseUri)))
+                            : messageListenerGenerator;
+                })
```

Comment:
Maybe this should be split into a method or methods so that it is a bit more readable.  It is not so easy to digest what is happening inside the flatmap